### PR TITLE
Adding the option public_ip that will allow to disable the external IP

### LINKED ISSUE #14: Make External IP configurable
For security reasons many customers only want to deploy VMs without external IP addresses. Currently this is not configurable in the scripts for HANA / HANA_HA. The NetWeaver scripts already have this config option named "external_ip". 

Is there a chance to add this to the HANA / HANA_HA setups? 
I'm happy to contribute if that makes sense for this little change.

### PR DESCRIPTION
**Pull request by michaeldtz**
I followed the already existing structures of the netweaver module and added the variable public_ip with the possible value 0 and 1. When 0 no external IP will be assigned. 

Fixes #14

### CODE REVIEW

**Comment by michaeldtz** [comment]
@morgante Happy to get your review. 

Please bear in mind that this is my first contribution. Sorry for any additional effort for you.

File: examples/sap_hana_ha_simple_example/variables.tf [hunk 1]
```diff
@@ -123,12 +123,19 @@ variable "network_tags" {
   default     = []
 }
 
+variable "public_ip" {
+  description = "Determines whether a public IP address is added to your VM instance."
+  default     = 1
```
Comment (morgante) [inline examples/sap_hana_ha_simple_example/variables.tf]: We should use `true` or `false` as values (and a `type = bool`).

File: modules/sap_hana/variables.tf [hunk 2]
```diff
@@ -114,8 +114,14 @@ variable "network_tags" {
   default     = []
 }
 
+variable "public_ip" {
+  description = "Determines whether a public IP address is added to your VM instance."
+  default     = 1
```
Comment (morgante) [inline modules/sap_hana/variables.tf]: This should be of type boolean.

File: modules/sap_hana/main.tf [hunk 3]
```diff
@@ -97,9 +99,13 @@ resource "google_compute_instance" "gcp_sap_hana" {
     subnetwork         = var.subnetwork
     subnetwork_project = var.project_id
 
-    access_config {
-      nat_ip = google_compute_address.gcp_sap_hana_ip.address
+    dynamic "access_config" {
+      for_each = var.public_ip == 1 ? google_compute_address.gcp_sap_hana_ip : []
+      content {
+        nat_ip = access_config.value.address
```
Comment (morgante) [inline modules/sap_hana/main.tf:105]: I don't think this is a valid value. What does it refer to?

**Review by morgante — CHANGES_REQUESTED** [review]
Thanks for working on this.

File: examples/sap_hana_ha_simple_example/variables.tf (hunk 1 above)
Comment (michaeldtz) [reply examples/sap_hana_ha_simple_example/variables.tf]: Thanks for the feedback. I'm happy to change this, I just had a look at the NetWeaver module before and that one is using an int and I was trying to make it the same behaviour. What do you think. Should we change it to bool here and accept the difference between the NW and HANA modules?

File: modules/sap_hana/main.tf (hunk 3 above)
Comment (michaeldtz) [reply modules/sap_hana/main.tf:105]: google_compute_address.gcp_sap_hana_ip refers to the external IP that gets created above. I added a count to that (above) and that makes it an array. That is why it works quite well with the for_each here.

So, yes I tested it and it is working properly.

Comment (morgante) [reply modules/sap_hana/main.tf:105]: Got it, I was having trouble following but that does indeed look quite sensible.

File: examples/sap_hana_ha_simple_example/variables.tf (hunk 1 above)
Comment (morgante) [reply examples/sap_hana_ha_simple_example/variables.tf]: Yeah I believe those modules were developed prior to Terraform v0.12's richer type support. Please go ahead and change it here - we can accept the difference between the modules.

Comment (michaeldtz) [reply examples/sap_hana_ha_simple_example/variables.tf]: +1 makes sense. I will adapt

**Comment by michaeldtz** [comment]
Changed the variable type to bool and adapted the conditional handling using count and for_each

**Comment by michaeldtz** [comment]
Ready for a new round of review

**Review by morgante — APPROVED** [review]

**Pull request merged**